ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs

```diff
@@ -229,10 +237,25 @@ where
     let pool_balance_key = balance_key(&nam(), &BRIDGE_POOL_ADDRESS);
     let relayer_rewards_key = balance_key(&nam(), relayer);
     // Remove the completed transfers from the bridge pool
-    for event in transfers {
+    for (event, is_valid) in
+        transfers.iter().zip(valid_transfers.iter().copied())
+    {
         let pending_transfer = event.into();
         let key = get_pending_key(&pending_transfer);
         if likely(wl_storage.has_key(&key)?) {
+            if likely(is_valid) {
```

Comment:
herm. This works because the rejected event will eventually time out, but it might be nicer is we de-escrow NAM / re-mint wrapped erc20s immediately.